### PR TITLE
websockets transport

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Client.hs [hunk 1]
```diff
@@ -92,7 +93,7 @@ data SMPClientConfig = SMPClientConfig
   { -- | size of TBQueue to use for server commands and responses
     qSize :: Natural,
     -- | default SMP server port if port is not specified in SMPServer
-    defaultPort :: ServiceName,
+    defaultTransport :: (ServiceName, ATransport),
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Client.hs:96]: update haskell doc

File: src/Simplex/Messaging/Client.hs [hunk 2]
```diff
@@ -126,88 +127,90 @@ type Response = Either SMPClientError Cmd
...
+      bSize <- tcpTimeout `timeout` atomically (takeTMVar thVar)
+      pure $ case bSize of
+        Just (Right blockSize) -> Right c {action, blockSize}
+        Just (Left e) -> Left e
+        Nothing -> Left SMPNetworkError
+
+    useTransport :: (ServiceName, ATransport)
+    useTransport = case port smpServer of
+      Nothing -> defaultTransport cfg
+      Just "80" -> ("80", transport @WS)
```
Comment (efim-poberezkin) [inline src/Simplex/Messaging/Client.hs:170]: I'm not sure about hardcoding port in `useTransport`.. why not have `Transport` as field of `SMPServer`?

**Review by efim-poberezkin — APPROVED** [review]

File: src/Simplex/Messaging/Client.hs (hunk 1 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Client.hs:96]: Which one?

File: src/Simplex/Messaging/Client.hs (hunk 2 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Client.hs:170]: Discussed, as port 80 can only be WS

**Pull request merged**